### DISCUSSION TITLE
Heavy Pages Lead to Slow Navigation Feedback

### DISCUSSION THREAD

**Discussion by rudrateja** [vercel/next.js#63612] [App Router]
We are experiencing significant delays when navigating between heavy pages in our Next.js application, particularly with Server-Side Rendered (SSR) and Incremental Static Regeneration (ISR) pages. 

I have created a sample project to reproduce this issue. check below link.

https://next-navigation-delay.vercel.app/

Expected Behavior:
Upon clicking a link, the user should immediately perceive feedback (visual, auditory, or haptic) indicating that the click was successful, followed by navigation to the linked content.

Actual Behavior:
Clicks on links do not always provide immediate feedback to the user, leading to confusion about whether the action was registered. The navigation eventually proceeds, but the lack of feedback is disconcerting.

**Comment by icyJoseph** [comment]
Hi,

Could you share the repository for this demo?

I had ran into this issue on my blog, I had written a 14 minutes long post with tons of code snippets. I'd click that post, and nothing would happen, for so long that I'd click again... and then it'd navigate...

What I did was wrap a suspense boundary around the code snippets, and that fixed it, or rather cut the time to navigate by a lot, it is still a bit slower than other posts tough (mostly because all code snippets, in all posts are wrapped now) ~ In this case that works, because my syntax highlighting engine blocks for too long. 

However, I wonder if you have a more reproducible demonstration, that I could study, in my case there was a lot of noise. Please do share a link to the repository if possible.

**Comment by rudrateja** [comment]
Hi @icyJoseph 

I hope this helps

https://github.com/rudrateja/next-navigation-delay

**Comment by bourbakiii** [comment]
Hi! Have you found a solution?
